pkg/ccl/backupccl/restore_job.go

```diff
@@ -289,39 +310,29 @@ func restore(
 		return emptyRowCount, err
 	}
 
+	//on231 := clusterversion.ByKey(clusterversion.V23_1Start).LessEq(job.Payload().
+	//	CreationClusterVersion)
+	on231 := true
```

Comment:
there's some unrelated bug where the commented out logic returns false on master. I need to investigate this. Perhaps the job object passed to `restore()` doesn't have the `CreationClusterVersion`.